lib/credit_note/credit_note.ex

```diff
@@ -0,0 +1,283 @@
+    """
+    @spec create([CreditNote.t() | map()], user: Project.t() | Organization.t() | nil) :: any
+    def create!(notes, options \\ []) do
+        Rest.post!(
+            resource(),
+            notes,
+            options
+        )
+    end
+
+    @doc """
+    # Retrieve a specific CreditNote
+    Receive a single CreditNote object previously created in the Stark Infra API by its id
+    ## Parameters (required):
+    `:id` [string]: object unique id. ex: "5656565656565656"
+    ## Parameters (optional):
+    `:user` [Organization/Project object, default nil]: Organization or Project object. Not necessary if StarkInfra.User was set before function call
+    ## Return:
+    `:CreditNote` object with updated attributes
+    """
+    @spec get(binary, user: Project.t() | Organization.t() | nil) ::
+        { :ok, [CreditNote.t()]} | {:error, [Error.t()]}
+    def get(id, options \\ []) do 
+        Rest.get_id(resource(), id, options)
+    end 
+
+    @doc """
+    Same as get(), but returns a CreditNote object instead of a list of errors
+    """
+    @spec get!(binary, user: Project.t() | Organization.t() | nil) :: any
+    def get!(id, options \\ []) do 
+        Rest.get_id!(resource(), id, options)
+    end
+    
+    @doc """
+    # Retrieve CreditNotes
+    Receive a generator of CreditNote objects previously created in the Stark Bank API
+    ## Parameters (optional):
+    `:limit` [integer, default nil]: maximum number of objects to be retrieved. Unlimited if nil. ex: 35
+    `:after` [Date or string, default nil] date filter for objects created only after specified date. ex: ~D[2020, 3, 10]
+    `:before` [Date or string, default nil] date filter for objects created only before specified date. ex: ~D[2020, 3, 10]
+    `:status` [string, default nil]: filter for status of retrieved objects. ex: "paid" or "registered"
+    `:tags` [list of strings, default nil]: tags to filter retrieved objects. ex: ["tony", "stark"]
+    `:ids` [list of strings, default nil]: list of ids to filter retrieved objects. ex: ["5656565656565656", "4545454545454545"]
+    `:user` [Organization/Project object, default nil]: Organization or Project object. Not necessary if StarkInfra.User was set before function call
+    ## Return:
+    `:generator` of CreditNote objects with updated attributes
+    """
+    @spec query(
+        limit: integer,
+        after: Date.t() | binary,
+        before: Date.t() | binary,
+        status: binary, 
+        tags: [binary],
+        ids: [binary],
+        user: Project.t() | Organization.t() | nil
+    ) :: 
+    ({:cont, {:ok, [CreditNote.t()]}}
+     | {:error, [Error.t()]}
+     | {:halt, any}
+     | {:suspend, any},
+      any -> 
+        any)
+    def query(options \\ []) do
+        Rest.get_list(resource(), options)
+    end
+
+    @doc """
+    Same as query(), but returns a generator of CreditNote objects instead of a list of errors
```

Comment:
Acho que o certo aqui seria isso:
Same as query(), but it will unwrap the error tuple and raise in case of errors.

Reply:
Aproveita pra conferir se nao entrou mais algum erro nesse estilo!